specs/core/1_shard-data-chains.md

```diff
@@ -243,15 +252,20 @@ def compute_crosslink_data_root(blocks: List[ShardBlock]) -> Hash:
             values += [b'\x00' * BYTES_PER_SHARD_BLOCK_BODY]
         return values
 
-    def merkle_root_of_bytes(data: bytes) -> bytes:
-        return merkle_root([data[i:i + 32] for i in range(0, len(data), 32)])
+    def hash_tree_root_of_bytes(data: bytes) -> bytes:
+        return hash_tree_root([data[i:i + 32] for i in range(0, len(data), 32)])
+
+    def zpad(data: bytes, length: int) -> bytes:
+        return data + b'\x00' * (length - len(data))
 
     return hash(
-        merkle_root(pad_to_power_of_2([
-            merkle_root_of_bytes(zpad(serialize(get_shard_header(block)), BYTES_PER_SHARD_BLOCK_BODY)) for block in blocks
-        ])) +
-        merkle_root(pad_to_power_of_2([
-                merkle_root_of_bytes(block.body) for block in blocks
+        hash_tree_root(pad_to_power_of_2([
+            hash_tree_root_of_bytes(
+                zpad(serialize(get_shard_header(block)), BYTES_PER_SHARD_BLOCK_BODY)
```

Comment:
I don't really know what `zpad` is supposed to do here. Based on the arguments, I implemented a `zpad` function above, but I am unsure whether this is correct.

More generally, I think the whole `compute_crosslink_data_root` function can be neatened up and simplified.